x-pack/plugins/index_management/public/application/components/component_templates/component_template_selector/component_templates.scss

```diff
@@ -22,10 +23,21 @@ $heightHeader: $euiSizeL * 2;
   &__searchBox {
     border-bottom: $euiBorderThin;
     border-top: $euiBorderThin;
+    border-top-right-radius: 0;
+    border-bottom-right-radius: 0;
     box-shadow: none;
     max-width: initial;
   }
 
+  &__filterListButton {
+    border-bottom: $euiBorderThin;
+    border-top: $euiBorderThin;
```

Comment:
nit: I would add also `border-left: $euiBorderThin;` to have a visual separation between the search box and the filter list button.

Reply:
Done 👍